easel/predicate.lua

```diff
@@ -1,44 +1,43 @@
 local M = {}
 
-
 local predicates = {}
-
-
-
--- create a predicate
--- this will cache the predicate and return the same one if requested more than
--- once
-function M.create(tags)
-	for i,tag in ipairs(tags) do
-		tag = type(tag) == "string" and hash(tag) or tag
-	end
-
+local function find_pred(hashed_list)
 	for _,predicate in ipairs(predicates) do
-		local found_predicate = true
-		for _,tag in ipairs(tags) do
-			if not predicate.tags_lut[tag] then
-				found_predicate = false
-				break
+        local found_predicate = false
+        if #predicate == #hashed_list then
+            found_predicate = true
+ 	    	for __,tag in ipairs(hashed_list) do
+		    	if not predicate.tags_lut[tag] then
+				    found_predicate = false
+				    break
+                end
 			end
 		end
-		if found_predicate then
-			return predicate.handle
-		end
-	end
-
-	local predicate = {
+		if found_predicate then return predicate end
+    end
+end
+local function add_pred(hashed_tags)
+      	local new_predicate = {
 		tags_lut = {},
-		handle = render.predicate(tags),
-	}
-	for _,tag in ipairs(tags) do
-		predicate.tags_lut[tag] = true
-	end
-
-	predicates[#predicates + 1] = predicate
-	return predicate.handle
+		handle = render.predicate(hashed_tags),
+    	}
+    	for _,tag in ipairs(hashed_tags) do
+	    	new_predicate.tags_lut[tag] = true
+    	end
+    	predicates[#predicates + 1] = new_predicate
 end
-
-
-
-
-return M
\ No newline at end of file
+ 
+function M.create(tags)
```

Comment:
The indentation is not consistent in this function. Could you please make sure that it follows the same style as the rest?

Reply:
Actually, it looks like the indentation has changed throughout the file. Please use tabs instead of spaces.

Reply:
Fixed now hopefully. I used vi to edit the file which I had set to replace tabs with spaces along time ago, this slipped my mind.